tikv: add the user's guide of stale read and safe ts (#14524)
pingcap/docs: TOC.md

    - [Identify Expensive Queries Using Top SQL](/dashboard/top-sql.md)
    - [Identify Expensive Queries Using Logs](/identify-expensive-queries.md)
    - [Save and Restore the On-Site Information of a Cluster](/sql-plan-replayer.md)

>     - [Understanding Stale Read and safe-ts in TiKV](/stale_read_user_guide.md)

Reviewer: ```suggestion
    - [Understanding Stale Read and safe-ts in TiKV](/troubleshoot-stale-read.md)
```

>     - [A User's Guide to Stale Read and Safe-ts in TiKV](/stale_read_user_guide.md)

Reviewer: ```suggestion
    - [Understanding Stale Read and safe-ts in TiKV](/stale_read_user_guide.md)
```
